src/diagnostic-channel-publishers/src/console.pub.ts

```diff
@@ -33,17 +33,26 @@ const consolePatchFunction: PatchFunction = (originalConsole) => {
 
         channel.publish<IConsoleData>("console", {message, stderr: true});
 
-        return process.stderr.write(chunk);
+        return true;
     };
 
     const aiLoggingConsole: Console = new originalConsole.Console(aiLoggingOutStream, aiLoggingErrStream);
-    aiLoggingConsole.Console = originalConsole.Console;
 
-    const consolePropertyDescriptor = Object.getOwnPropertyDescriptor(global, "console");
-    consolePropertyDescriptor.get = function() { return aiLoggingConsole; };
-    Object.defineProperty(global, "console", consolePropertyDescriptor);
+    const consoleMethods = ["log", "info", "warn", "error", "dir", "time", "timeEnd", "trace", "assert"];
 
-    return aiLoggingConsole;
+    for (const method of consoleMethods) {
+        const originalMethod = originalConsole[method];
+        if (originalMethod) {
+            originalConsole[method] = function() {
+                if (aiLoggingConsole[method]) {
+                    aiLoggingConsole[method].apply(aiLoggingConsole, arguments);

```

Comment:
Maybe try catch, just in case the arguments aren't what we expect? Or should it never throw regardless?